models.py

```diff
@@ -1,7 +1,137 @@
-import threading
+"""
+Models for Inventory Management Service
+
+All of the models are stored in this module
+
+Models
+------
+Entry - An Inventory Entry used in the service
+
+Attributes:
+-----------
+prod_id         (int)       - the id of the product
+prod_name       (string)    - the name of the product
+new_qty         (int)       - quantity of the new product
+used_qty        (int)       - quantity of the used product
+open_boxed_qty  (int)       - quantity of the open-boxed product
+restock_level   (int)       - when the product total quantity reaches below this number,
+                              an automatic restock will be trigger.
+restock_amt     (int)       - the amount of new products restocked
+                              when the total quantity goes under restock_level
+
+"""
+
+# import threading
+from flask_sqlalchemy import SQLAlchemy
 
 class DataValidationError(Exception):
-	pass
+    """ Used for an data validation errors when deserializing """
+    pass
+
+db = SQLAlchemy()
+
+class Entry(db.Model):
+    """ A class representing an Inventory entry"""
+
+    app = None
+
+    # Table Schema
```

Comment:
Should these fields be instance-based rather than class-based?

Also, I think it is better to abstract the table schema into another class and create an instance of it in the init function when instantiating an Entry class.

Reply:
Hi, since I'm new to SQLAlchemy (& python) and referred to prof's code when writing this part, I can only make my guess about why this code should be like this, I would suggest you to relay this question to professor if you're interested in a more precise answer. Plus I'm also interest in what prof would say.

The Entry class (renamed to ProductInformation now) is connected to an inner database, and the "app" is class-based so that every time we created a new instance of Entry class we don't have to "make the connection" with the "app" again.

Besides, the Entry class inherited from "db.model" which makes it more like a db rather than a class for enclosing a certain product info (noticing the column definitions inside Entry class). But it can somehow also work as a class for enclosing a certain product's info. I'm guessing it is a privilege from the library so we don't have to create another class for simple database queries.

Reply:
This seems to be weird. I'll unblock you now but I think this is worth investigation.